### ISSUE TITLE
Add KindleGen and Kindle dev/build/publish scripts

### ISSUE THREAD

**Issue by thephilgray** [thephilgray/gulp-epub-starter#5]
Add KindleGen and Kindle dev/build/publish scripts

**Comment by thephilgray** [comment]
Not sure it's acceptable to redistribute the kindlegen software.

**Comment by thephilgray** [comment]
Also, is there a possible conflict if a rendition should be both reflowable and kindle?

**Comment by thephilgray** [comment]
1. update `tasks/assets.js` to filter out `cover.pug` if `DEVICE === 'kindle'`

This is dangerous, if the cover page template needs to be named something else. Add cover page path to config or generate it outside of pages.

```js
const filter = require("gulp-filter");

export const pages = () => {
  const f = filter(["**", "!**/**/cover.pug"]);
  let currentPageNumber = 0;
  return (
    gulp
      .src(["./src/pages/**/*.pug"], { base: "./src/pages/" })
      /* filter out cover.pug for kindle */
      .pipe(gulpif(DEVICE === "kindle", f))
      .pipe(
        data(file => ({
          filename: path.basename(file.path).replace(".pug", ""),
          pageNumber: ++currentPageNumber
        }))
      )
      .pipe(
        pug({
          doctype: "xhtml",
          locals: {
            ...settings,
            viewport: settings.devices[DEVICE].viewport,
            fixed: FIXED,
            device: DEVICE
          }
        })
      )
      .pipe(
        gulpif(
          DEVELOPMENT,
          htmltidy({
            doctype: "xhtml",
            outputXhtml: "yes",
            indent: "auto",
            wrap: 120,
            "wrap-attributes": false,
            "drop-empty-elements": "no"
          })
        )
      )
      .pipe(
        gulpif(
          PRODUCTION,
          htmltidy({
            inputXml: true,
            outputXhtml: true,
            indent: "auto",
            wrap: 120,
            "wrap-attributes": false,
            "drop-empty-elements": "no"
          })
        )
      )
      .pipe(extReplace(".xhtml"))
      .pipe(gulp.dest(contentDir + "/xhtml"))
  );
};
```

2. update `tasks/index.js` to add a a kindlegen task and a new publish task that includes kindlegen at the end

Danger: requires `kindlegen` to be installed on the user's local machine. This needs to be documented.

```js
/* requires kindlegen installed locally */
export const kindlegen = done => {
  exec(`kindlegen ./builds/${epubName}`, (err, stdout, stderr) => {
    if (err) {
      console.error(err);
    }
    console.log(stdout);
  });
  done();
};

export const publish = gulp.series(build, zipEpub, validate);

export const publishKindle = gulp.series(publish, kindlegen);
```
3. Update npm scripts at `package.json` to include dev/build/publish tasks for DEVICE=kindle

```json
{
 "dev:kindle": "DEVICE=kindle NODE_ENV=development gulp --require @babel/register --gulpfile tasks",
dev:kindle:fixed": "DEVICE=kindle NODE_ENV=development gulp --require @babel/register --gulpfile tasks --fixed",
"build:kindle": "DEVICE=kindle NODE_ENV=production gulp build --require @babel/register --gulpfile tasks",
"build:kindle:fixed": "DEVICE=kindle NODE_ENV=production gulp build --require @babel/register --gulpfile tasks --fixed",
"publish:kindle": "DEVICE=kindle NODE_ENV=production gulp publishKindle --require @babel/register --gulpfile tasks",
"publish:kindle:fixed": "DEVICE=kindle NODE_ENV=production gulp publishKindle --require @babel/register --gulpfile tasks --fixed",
}

```

**Comment by thephilgray** [comment]
closed by #51
